roles/ovirt-engine-setup/tasks/main.yml

```diff
@@ -0,0 +1,72 @@
+---
+# health page
+- block:
+  - name: check if ovirt-engine running (health page)
+    uri:
+      url: "http://{{ ansible_fqdn }}/ovirt-engine/services/health"
+      status_code: 200
+    register: ovirt_engine_status
+    retries: 2
+    delay: 5
+    until: ovirt_engine_status|success
+    ignore_errors: True
+
+  # copy default answer file
+  - name: copy default answerfile
+    template:
+      src: answerfile_{{ ovirt_engine_version }}_basic.txt.j2
+      dest: /tmp/answerfile.txt
+      mode: 0644
+      owner: root
+      group: root
+    when: ovirt_engine_answer_file_path is undefined
+
+  # copy custom answer file
+  - name: copy custom answer file
+    no_log: True
+    template:
+      src: "{{ ovirt_engine_answer_file_path }}"
+      dest: /tmp/answerfile.txt
+      mode: 0600
+      owner: root
+      group: root
+    when: ovirt_engine_answer_file_path is defined
+
+  - name: update setup packages
+    yum:
+      name: 'ovirt*setup*'
+      state: latest
+    when: ovirt_engine_update
+    tags:
+      - skip_ansible_lint
+
+  - name: run engine-setup with answerfile
+    shell: 'engine-setup --config-append=/tmp/answerfile.txt'
```

Comment:
What is the reason to use `shell`, instead of `command`? It is a best practise to use `command` unless `shell` is a must to use.

Reply:
Not sure it's a valid reason, but you might want to pass certain shell env vars, e.g. 'OTOPI_DEBUG=1'. Not sure ansible's shell modules allows to do this without changing the play either.